src/modules/load_mon/load_mon.cpp

```diff
@@ -285,20 +281,22 @@ void LoadMon::_stack_usage()
 		perf_end(_stack_perf);
 
 		if (checked_task) {
-			/*
-			 * Found task low on stack, report and exit. Continue here in next cycle.
-			 */
-			if (stack_free < 300) {
-				strncpy((char *)_low_stack.task_name, system_load.tasks[task_index].tcb->name, low_stack_s::MAX_REPORT_TASK_NAME_LEN);
-				_low_stack.stack_free = stack_free;
 
-				if (_low_stack_pub == nullptr) {
-					_low_stack_pub = orb_advertise(ORB_ID(low_stack), &_low_stack);
+			_task_stack_info.stack_free = stack_free;
+			_task_stack_info.timestamp = hrt_absolute_time();
 
-				} else {
-					orb_publish(ORB_ID(low_stack), _low_stack_pub, &_low_stack);
-				}
+			if (_task_stack_info_pub == nullptr) {
+				_task_stack_info_pub = orb_advertise_queue(ORB_ID(task_stack_info), &_task_stack_info, num_tasks_per_cycle);
 
+			} else {
+				orb_publish(ORB_ID(task_stack_info), _task_stack_info_pub, &_task_stack_info);
+			}
+
+			/*
+			 * Found task low on stack, report and exit. Continue here in next cycle.
+			 */
+			if (stack_free < STACK_LOW_WARNING_THRESHOLD) {
+				PX4_WARN("%s low on stack! (%i bytes left)", _task_stack_info.task_name, stack_free);
```

Comment:
we do save to .ulog and the web graphing tool will display it. I think that's better than spamming the console.